gbn/queue.go

```diff
@@ -219,6 +468,45 @@ func (q *queue) processNACK(seq uint8) (bool, bool) {
 	return true, bumped
 }
 
+// proceedAfterTime will wait for the resendTimeout and then send an
+// awaitedACKSignal, if we're still awaiting the resend catch up.
+func (q *queue) proceedAfterTime(resendTimeout time.Duration) {
+	go func() {
+		ticker := time.NewTicker(resendTimeout)
```

Comment:
Same here re using the tiny bit more lightweight `time.After()`, since we're never resetting the timer.